File: malai-core/org.malai.core/src/test/java/org/malai/fsm/StubTransitionOK.java
package org.malai.fsm;

import java.util.Set;
import org.mockito.internal.util.collections.Sets;

public class StubTransitionOK extends Transition<StubEvent> {
	protected boolean guard;

	protected StubTransitionOK(final OutputState<StubEvent> srcState, final InputState<StubEvent> tgtState) {
		this(srcState, tgtState, true);
	}

	protected StubTransitionOK(final OutputState<StubEvent> srcState, final InputState<StubEvent> tgtState, final boolean guard) {
		super(srcState, tgtState);
		this.guard = guard;
	}

	@Override
	protected boolean accept(final StubEvent event) {
		return true;
	}

	@Override
	protected boolean isGuardOK(final StubEvent event) {
		return guard;
	}

	@Override
	public Set<Object> getAcceptedEvents() {
		return Sets.newSet(StubEvent.class);
	}
}

class SubStubTransition1 extends StubTransitionOK {
	protected SubStubTransition1(final OutputState<StubEvent> srcState, final InputState<StubEvent> tgtState, final boolean guard) {
		super(srcState, tgtState, guard);
	}

	@Override
	public boolean accept(final StubEvent event) {
		return event instanceof StubSubEvent1;
	}

	@Override
	public Set<Object> getAcceptedEvents() {
		return Sets.newSet(StubSubEvent1.class);
	}
}

class SubStubTransition2 extends StubTransitionOK {
	protected SubStubTransition2(final OutputState<StubEvent> srcState, final InputState<StubEvent> tgtState, final boolean guard) {
		super(srcState, tgtState, guard);
	}

	@Override
	public boolean accept(final StubEvent event) {
		return event instanceof StubSubEvent2;
	}

	@Override
	public Set<Object> getAcceptedEvents() {
		return Sets.newSet(StubSubEvent2.class);
	}
}

class SubStubTransition3 extends StubTransitionOK {
	protected SubStubTransition3(final OutputState<StubEvent> srcState, final InputState<StubEvent> tgtState, final boolean guard) {
		super(srcState, tgtState, guard);
	}

	@Override
	public boolean accept(final StubEvent event) {
		return event instanceof StubSubEvent3;
	}

	@Override
	public Set<Object> getAcceptedEvents() {
		return Sets.newSet(StubSubEvent3.class);
	}
}

File: malai-core/org.malai.ts/test/fsm/StubTransitionOK.ts
import {Transition} from "../../src-core/fsm/Transition";
import {StubEvent, StubSubEvent1, StubSubEvent2, StubSubEvent3} from "./StubEvent";
import {InputState} from "../../src-core/fsm/InputState";
import {OutputState} from "../../src-core/fsm/OutputState";

export class StubTransitionOK extends Transition<StubEvent> {
    public guard: boolean;

    public constructor(srcState: OutputState<StubEvent>, tgtState: InputState<StubEvent>, guard ?: boolean) {
        super(srcState, tgtState);
        this.guard = guard === undefined ? true : guard;
    }

    public accept(event: StubEvent): boolean {
        return true;
    }

    public isGuardOK(event: StubEvent): boolean {
        return this.guard;
    }

    public getAcceptedEvents(): Set<string> {
        return new Set(["StubEvent"]);
    }
}

export class SubStubTransition1 extends StubTransitionOK {
    public constructor(srcState: OutputState<StubEvent>, tgtState: InputState<StubEvent>, guard ?: boolean) {
        super(srcState, tgtState, guard);
    }

    public accept(event: StubEvent): boolean {
        return event instanceof StubSubEvent1;
    }

    public getAcceptedEvents(): Set<string> {
        return new Set("StubSubEvent1");
    }
}

export class SubStubTransition2 extends StubTransitionOK {
    public constructor(srcState: OutputState<StubEvent>, tgtState: InputState<StubEvent>, guard ?: boolean) {
        super(srcState, tgtState, guard);
    }

    public accept(event: StubEvent): boolean {
        return event instanceof StubSubEvent2;
    }

    public getAcceptedEvents(): Set<string> {
        return new Set("StubSubEvent2");
    }
}

export class SubStubTransition3 extends StubTransitionOK {
    public constructor(srcState: OutputState<StubEvent>, tgtState: InputState<StubEvent>, guard ?: boolean) {
        super(srcState, tgtState, guard);
    }

    public accept(event: StubEvent): boolean {
        return event instanceof StubSubEvent3;
    }

    public getAcceptedEvents(): Set<string> {
        return new Set("StubSubEvent3");
    }
}
